core/types/subtyping.cc

```diff
@@ -927,7 +927,8 @@ void compareToUntyped(Context ctx, TypeConstraint &constr, const TypePtr &ty, co
 
 // "Single" means "ClassType or ProxyType"; since ProxyTypes are constrained to
 // be proxies over class types, this means "class or class-like"
-bool isSubTypeUnderConstraintSingle(Context ctx, TypeConstraint &constr, const TypePtr &t1, const TypePtr &t2) {
+bool isSubTypeUnderConstraintSingle(Context ctx, TypeConstraint &constr, const bool allowUntyped, const TypePtr &t1,
```

Comment:
`const bool`? :-)
Sure :-)

Reply:
Happy to change it, but after writing haskell for so long, I'm terrified of spurious mutations :)